### ISSUE TITLE
Give guests access to specific channels

### ISSUE THREAD

**Issue by designgurudotorg** [RocketChat/Rocket.Chat#6267]
Is it currently possible to allow guests to access a specific channel by opening up a public URL to it? Haven't seen a system setting anywhere to facilitate this and am looking for a way to open up RC to the public for specific conversations/channels for a limited amount of time (so you could toggle the public access permissions to a channel on or off say)

q./

**Comment by engelgabriel** [comment]
Related to https://github.com/RocketChat/Rocket.Chat/issues/1848

**Comment by icerootsvw** [comment]
Are there any updates in this feature request?

**Comment by piotrkochan** [comment]
I'm looking forward for this too.

**Comment by sandys** [comment]
Is there any update here ? We are looking to move from hipchat and looking for similar features.

Hipchat had guest access and so does slack. It is is very very important feature for teams that work with consultants,etc.

**Comment by Giplat** [comment]
Hi, any updates on adding guests to channels? Having 'unlimited guests' is prominently featured on the Rocketchat website, but I can't find any documentation on how to handle guest users.

**Comment by sitzmoebelchronograph** [comment]
Hi guys,

are any updates available for this?

I think it's an important feature for use cases, if you want to have a separated chat room for each of your customers with guest accounts.

There is a function in HipChat to enable guest access for people by presenting a unique URL for each time you enable guest access. A customer visit the URL and is able to type in a username and joins the specified room.

A feature like this would pretty cool to have in RocketChat.

Enabling guest access and all guests are able see all public rooms and also able to join them.
This is not the same like, enabling guest access for a specified room.

Greets
Andreas

**Comment by icerootsvw** [comment]
From my point of view Rocket Chat has a very powerful guest feature, just a little bit of configuration is needed.

I had to create guest users and granting the "guest" role to them. After that i was revoking some access rights (see other channels, only see channels you are invited too).

![image](https://user-images.githubusercontent.com/40857785/60414796-8b5f5500-9bd9-11e9-9bd3-874bc55bad03.png)


Of course still an existing user is needed but we are fine atm with this feature, so we can add external persons to our chat system and they can only see channels designed for them

**Comment by FuniCelas** [comment]
Hey :) 

is there any progress for this feature? 

I am considering to implement rocket.chat in my company. However, it is imperative that we can communicate with partners from outside.

**Comment by icerootsvw** [comment]
@FuniCelas did you had a look at my comment? We are using guest users here

**Comment by FuniCelas** [comment]
> @FuniCelas did you had a look at my comment? We are using guest users here

Yes, I saw it. However, this is not the solution to the request. It was about being able to send a link to an external partner, who would then have immediate access to the chat, etc. This function is important to me, because it simplifies a lot before I have to create a new user every time or explain to the external partner how to register as a guest.

So can I assume that this function is not planned for the future?

**Comment by rotanid** [comment]
we did set up a separate rocket.chat instance just for communication with externals, because this feature is missing :-(

**Comment by sampaiodiego** [comment]
on version 3.2.0 we have updated the guest feature to make sure they can only see the rooms they're invited to, they also cannot start a DM with someone.

guest users still need to be created manually via admin panel by giving them only the "guest" role.

please let me know if this is what you guys are expecting.. thx

**Comment by pgress** [comment]
If this is some "make a wish"-thing: I create a channel for guests and get a shareable link to this room. They can use the link and give themself a name. Normal users can join the usual way. The users who are created by this mechanism are deleted periodically by some garbage collector (or something like that)

**Comment by sampaiodiego** [comment]
thanks @pgress 😂  this is not how it works right now.. but we have discussed this while creating the invite links feature.. having an option while creating an invite link to make users registered with that link expire after a period of time.. this will be an addition to "invite links", to possibly be used in conjunction with "guests"..

**Comment by rotanid** [comment]
thanks for the ongoing work on this, and i second what @pgress wrote
